Tensile/Source/client/include/HardwareMonitor.hpp

```diff
@@ -143,6 +152,9 @@ namespace Tensile
 
             std::vector<uint32_t> m_fanMetrics;
             std::vector<int64_t>  m_fanValues;
+            std::vector<uint16_t> m_freqValues;
```

Comment:
I read the existing code.
It looks like you implemented these vectors in a different way compared to other existing metrices.
Other existing vectors represent different device/HW, but you use these vectors to store multiple inputs.
It is probably because of median, right?
Why we use median only for new metrices?

The current way is a bit confusing.

Reply:
Yes. Initially I thought to keep the same old way. but I wanted to print all these information at the end of kernel execution. do you think these information will not be useful later ?
GFX Value				PPT0_value	       Temperature
1262         				190					36
1258         				140					35
1258         				140					35
1258         				140					35
1258         				138					36
1259         				190					36
1259         				190					36
1259         				236					36
1262         				236					36
1262         				236					36
1265         				256					36
1265         				256					36
1265         				256					36
1272         				261					36
1272         				261					36
1301         				261					36
1272         				263					36
1288         				263					36
1288         				263					36
1288         				265					36

				Min	   Max	Average	Median
GFX Frequency	1258    1301    1268.55    1265    
Power Value		  138     265     222.05      256     
Temperature		  35      36      35.85          36

Reply:
We can keep track of max/min value by adding variables for them.
The only thing is median.
If we really need it, I am OK, but I suggest to use different naming for these vector variables and/or put some comment to recognize the new different way.

Reply:
Or, use vector of vector to make the structure align with others.

Reply:
Added comments to recognize new different way. I tried to change to current implementation to match existing  but realized it is not needed hence not changing the implementation. Median value is needed and also better to preserve the individual data values. 
Current implementation uses ROCm API (rsmi_dev_gpu_metrics_info_get) where we get all HW type details through single invocation. Existing metrics implementation multiple iteration to store different HW type. if we need to store different HW type in the future, new additonal vector type or vector<vector>>would be appropriate in future.